src/helpers/get-insights-anonymous-user-token.ts

```diff
@@ -0,0 +1,21 @@
+export const ANONYMOUS_TOKEN_COOKIE_KEY = '_ALGOLIA';
+
+function getCookie(name: string): string {
+  const prefix = `${name}=`;
+  const cookies = document.cookie.split(';');
+  for (let i = 0; i < cookies.length; i++) {
+    let cookie = cookies[i];
+    while (cookie.charAt(0) === ' ') {
+      cookie = cookie.substring(1);
+    }
+    if (cookie.indexOf(prefix) === 0) {
+      return cookie.substring(prefix.length, cookie.length);
+    }
+  }
+  return '';
+}
+
+export default function getInsightsAnonymousUserToken(): string | null {
+  const anonymousUserToken = getCookie(ANONYMOUS_TOKEN_COOKIE_KEY);
+  return anonymousUserToken || null;
```

Comment:
I'd keep it as undefined in the return, I think the Algolia API will throw if you pass it `null`, but it will be fine on `undefined`, since that doesn't get serialised.

Reply:
same here, returning `null` here instead of `undefined` because  ESlint rule `consistent-return`.

Reply:
fixed here aee4083